docs/web/core-concepts/modules.mdx

```diff
@@ -0,0 +1,419 @@
+      return await dbTodos.findById(id);
+    },
+
+    // Get all todos for the current user
+    async getAll({}, { user }) {
+      return await dbTodos.fetch({ userId: user.id });
+    },
+
+    // Get todos with filtering
+    async getCompleted({}, { user }) {
+      return await dbTodos.fetch({
+        userId: user.id,
+        isCompleted: true
+      });
+    }
+  }
+});
+```
+
+### Query Parameters
+
+Queries receive two arguments:
+
+1. **Input parameters** - Data passed from the client
+2. **Context** - Server-side context including:
+   - `user` - Current authenticated user (if logged in)
+   - `req` - Express request object
+   - `res` - Express response object
+
+### Calling Queries from the Client
+
+```typescript
+import { callMethod } from 'modelence/client';
+
+// Call a query
+const todos = await callMethod('todo.getAll');
+
+// Call with parameters
+const todo = await callMethod('todo.getOne', { id: '123' });
+```
+
+With React Query:
+
+```typescript
+import { useQuery } from '@tanstack/react-query';
+import { modelenceQuery } from '@modelence/react-query';
+
+function TodoList() {
+  const { data: todos } = useQuery(
+    modelenceQuery('todo.getAll')
+  );
+
+  return <div>{/* render todos */}</div>;
+}
+```
+
+## Mutations
+
+Mutations are functions that modify data. They're similar to POST/PUT/DELETE endpoints in REST or mutations in GraphQL.
+
+```typescript
+export default new Module('todo', {
+  mutations: {
+    // Create a new todo
+    async create({ title, dueDate }, { user }) {
+      const { insertedId } = await dbTodos.insertOne({
+        title,
+        dueDate,
+        userId: user.id,
+        isCompleted: false,
+        createdAt: new Date()
+      });
+      return insertedId;
+    },
+
+    // Update a todo
+    async update({ id, title, isCompleted }) {
+      return await dbTodos.updateOne(
+        { id },
+        { $set: { title, isCompleted } }
+      );
+    },
+
+    // Delete a todo
+    async delete({ id }) {
+      return await dbTodos.deleteOne({ id });
+    }
+  }
+});
+```
+
+### Calling Mutations from the Client
+
+```typescript
+import { callMethod } from 'modelence/client';
+
+// Call a mutation
+const todoId = await callMethod('todo.create', {
+  title: 'Buy groceries',
+  dueDate: new Date('2024-12-31')
+});
+```
+
+With React Query:
+
+```typescript
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { modelenceMutation } from '@modelence/react-query';
+
+function CreateTodo() {
+  const queryClient = useQueryClient();
+
+  const { mutate: createTodo } = useMutation({
+    ...modelenceMutation('todo.create'),
+    onSuccess: () => {
+      // Invalidate and refetch todos after creation
+      queryClient.invalidateQueries({ queryKey: ['todo.getAll'] });
+    }
+  });
+
+  const handleSubmit = () => {
+    createTodo({ title: 'New todo', dueDate: new Date() });
+  };
+
+  return <button onClick={handleSubmit}>Create Todo</button>;
+}
+```
+
+## Stores
```

Comment:
Maybe it's better to talk about Stores first and then about Mutations and Queries as all examples use dbTodos, which is defined only here in Stores.

Reply:
Done